.github/workflows/build.yml

```diff
@@ -0,0 +1,41 @@
+name: Build Wiresmash with CMake
+
+on:
+  push:
```

Comment:
Just to be sure : this is on which branch the test will run, when Github detects a push/a pull request?

Reply:
let's explain everything, shall we? :yum: 

- the `on` field specifies the events on which the pipeline is automatically triggered
- nowadays, there is only the `main` branch and it is push-protected, meaning that we can not manually push to it and the settings require to go through a review process and a PR :ok_hand: 
- the `on.pull_request.branches` field gives the list of all the branches that will trigger a CI job when a PR is opened on them
for now, it's obviously only the `main` branch
- the `on.push.branches` is quite the same but for direct pushes to the branches in the list, i.e. the `main` branch only for now

## an example to wrap this up!
- i have a bunch of commit on the `main` branch
- i try to push them on `iScsc/wiresmash/main`, but this is refused thanks to the branch protection
- so i create a `feature` branch, push to either `iSCsc/wiresmash` or a fork of mine
- i open a PR onto `iScsc/wiresmash/main`
- as this is part of `on.pull_request.branches` a CI job is triggered in the PR body
this job can, when properly configured, block a PR until it does not fail anymore, very handy to block a failing source base :ok_hand: 
- the review process goes on
- we merge the PR -> **this is a push on the `main` branch** so, as it is part of `on.push.branches`, this triggers a CI job on the `main` branch, appearing on the front page of the repo, with a :red_circle:, a :orange_circle: or a :green_circle:  depending on the status of the last CI job :star_struck: 

## about the new branches
yes, depending on the new branches and their meaning and use, we will add them somewhere in the `on` field and in the branch protection rules list :+1:

Reply:
Ty for the explanation, I got the gist of it

Reply:
greeeat :relieved: :muscle: